api/article/views.py

```diff
@@ -17,3 +34,15 @@ def perform_create(self, serializer, format=None):
             serializer.save(author=author, image=image)
         else:
             serializer.save(author=author)
+
+    def perform_update(self, serializer):
+        if self.request.data.get('image') is not None:
+            image = self.request.data.get('image')
+            serializer.save(image=image)
+        else:
+            serializer.save()
+
+    # TODO permissions author only
+    @detail_route(methods=['get'], permission_classes=[permissions.IsAuthenticated])
```

Comment:
Also you can manage you `permission_classes`  using them as `property`.
```python
@property
def  permission_classes(self):
    if self.action == 'self_articles':
        return articles.filter(author=author)
    return foo
```

Reply:
I plan add a custom permissions later